src/App.js

```diff
@@ -5,12 +5,30 @@ import Carrinho from './pages/Carrinho';
 import Home from './pages/Home';
 
 class App extends React.Component {
+  constructor() {
+    super();
+
+    this.state = {
+      carrinho: [],
+    };
+  }
+
+  addAoCarrinho = (produto) => {
+    this.setState((state) => ({ carrinho: [...state.carrinho, produto] }));
```

Comment:
```suggestion
    this.setState(({carrinho}) => ({ carrinho: [...carrinho, produto] }));
```
Opa Diego, seria possível desestruturar no setState o estado, assim deixando o codigo mais enxuto, concordas?